### ISSUE TITLE
SelectPoller._flush_outbound would crash due to missing _manage_event_state member

### ISSUE THREAD

**Issue by vitaly-krugl** [pika/pika#565]
This concerns changes in PR #559: SelectPoller._flush_outbound would crash due to missing _manage_event_state member

```
    def _flush_outbound(self):
        """Call the state manager who will figure out that we need to write
            then call the poller's poll function to force it to process events.
        """
        self._manage_event_state()
        # Force our poller to come up for air, but in write only mode
        # write only mode prevents messages from coming in and kicking off
        # events through the consumer
        self.poll(write_only=True)
```
If confirmed, this code path could also benefit from a test.

I checked, and nothing accesses `_flush_outbound()` on any of the pollers, so it should be okay to just delete `SelectPoller._flush_outbound` as a better alternative to fixing it:
```
$ git grep _flush_outbound
pika/adapters/base_connection.py:    def _flush_outbound(self):
pika/adapters/blocking_connection.py:        self._flush_outbound()
pika/adapters/blocking_connection.py:    def _flush_outbound(self):
pika/adapters/select_connection.py:    def _flush_outbound(self):
pika/adapters/select_connection.py:    def _flush_outbound(self):
pika/adapters/twisted_connection.py:        self._flush_outbound call that was in _send_frame which previously
pika/connection.py:    def _flush_outbound(self):
pika/connection.py:        self._flush_outbound()
pika/connection.py:            self._flush_outbound()
tests/unit/connection_tests.py:        self.connection._flush_outbound = mock.Mock()
tests/unit/connection_tests.py:        self.connection._flush_outbound = mock.Mock()
```

**Comment by vitaly-krugl** [comment]
CC @wjps

**Comment by wjps** [comment]
Fixed in #578

**Comment by vitaly-krugl** [comment]
Fixed, thanks
